src/Storefront/Resources/app/storefront/src/plugin/address-manager/address-manager.plugin.js

```diff
@@ -0,0 +1,249 @@
+import Plugin from 'src/plugin-system/plugin.class';
+import HttpClient from 'src/service/http-client.service';
+import PseudoModalUtil from 'src/utility/modal-extension/pseudo-modal.util';
+import ButtonLoadingIndicatorUtil from 'src/utility/loading-indicator/button-loading-indicator.util';
+
+const SHIPPING = 'shipping';
+const BILLING = 'billing';
+
+/**
+ * @package checkout
+ */
+export default class AddressManagerPlugin extends Plugin {
+
+    static options = {
+        activeBillingAddressId: null,
+        activeShippingAddressId: null,
+        initialTab: null,
+        searchSelectorShipping: '.address-manager-modal-list-search-shipping',
+        searchSelectorBilling: '.address-manager-modal-list-search-billing',
+        addressEditForm: '.address-manager-modal-address-form',
+        shippingAddressTabId: 'shipping-address-tab',
+        billingAddressTabId: 'billing-address-tab',
+        shippingTabPaneId: 'shipping-address-tab-pane',
+        billingTabPaneId: 'billing-address-tab-pane',
+        addressManagerModalSelector: '.address-manager-modal',
+        addressEditCancelSelector: '.address-form-create-cancel',
+        addressEditSubmitSelector: '.address-form-create-submit',
+        setDefaultAddress: '.address-manager-modal-set-default-address',
+        addressEditorFormSubmit: '#address-manager-modal-address-form',
+        currentShippingId: '.address-manager-modal-currentShippingId',
+        currentBillingId: '.address-manager-modal-currentBillingId',
+        addressSelectItem: '.address-manager-modal-select-address',
+    };
+
+    init() {
+        if (![SHIPPING, BILLING].includes(this.options.initialTab)) {
+            console.warn(`[AddressManagerPlugin] options.initialTab was expected to be 'billing' or 'shipping', got '${this.options.initialTab}'`);
+            this.options.initialTab = SHIPPING;
+        }
+
+        const onClick = this._getModal.bind(this);
+
+        this.el.removeEventListener('click', onClick);
+        this.el.addEventListener('click', onClick);
```

Comment:
not really necessary to dedupe
```suggestion
        this.el.removeEventListener('click', this._getModal.bind(this));
        this.el.addEventListener('click', this._getModal.bind(this));
```